### PR TITLE
Migrate template report to form

### PR DESCRIPTION
**Pull request by GretaD**
ref https://github.com/nextcloud/groupware/issues/13

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
+name: "Calendar Bug"
+description: "Have you encountered a bug?"
+labels: ["bug", "0. Needs triage"]
```
Comment (ChristophWurst) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:3]: compare with https://github.com/nextcloud/calendar/labels. the triage label has a different name.

Comment (GretaD) [reply .github/ISSUE_TEMPLATE/bug_report_form.yml:3]: ok, anything else to change?

**Review by ChristophWurst — APPROVED** [review]
Looks good. Do you have a live example in a test repo?

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+description: "Have you encountered a bug?"
+labels: ["bug", "0. to triage"]
+body:
+  - type: textarea
+    id: reproduce
+    attributes:
+      label: Steps to reproduce
+      description: |
+        Describe the steps to reproduce the bug.
+        The better your description is _(go 'here', click 'there'...)_ the fastest you'll get an _(accurate)_ answer.
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:11]: I'm just being pedant
```suggestion
        The better your description is _(go 'here', click 'there'…)_ the fastest you'll get an _(accurate)_ answer.
```

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      description: |
+        Tell us what happens instead
+    validations:
+      required: true
+  - type: textarea
+    id: calendar-app-version
+    attributes:
+      label: calendar app version
+      description: |
+        See apps admin page, e.g. 0.5.3
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:38]: Let's give an example value more recent :)

**Review by tcitworld — COMMENTED** [review]
So in the current bug template [we have a comment](https://github.com/nextcloud/calendar/blob/1aadc6101ea1dcea578bce1e7c626ddaef911b79/.github/ISSUE_TEMPLATE/bug.md?plain=1#L10-L24) that gives a number of informations, notably that you should post your CalDAV-related issues on the server repo instead. How do we give these informations with this system?

**Comment by ChristophWurst** [comment]
> So in the current bug template [we have a comment](https://github.com/nextcloud/calendar/blob/1aadc6101ea1dcea578bce1e7c626ddaef911b79/.github/ISSUE_TEMPLATE/bug.md?plain=1#L10-L24) that gives a number of informations, notably that you should post your CalDAV-related issues on the server repo instead. How do we give these informations with this system?

According to https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

```yml
body:
  - type: markdown
    attributes:
      value: |
        Thanks for reporting issues back to Nextcloud! This is the issue tracker of Nextcloud, if you have any support question please check out https://nextcloud.com/support
        This is the bug tracker for the web-calendar component. Find other components at https://github.com/nextcloud/
        Everything that is related to the CalDAV server, meaning it does not only affect the web-calendar but also your mobile and desktop clients, belongs in the Server repository: https://github.com/nextcloud/server
        For reporting potential security issues please see https://nextcloud.com/security/
        To make it possible for us to help you please fill out below information carefully. 
        You can also use the Issue Template application to prefill most of the required information: https://apps.nextcloud.com/apps/issuetemplate
        If you are a customer, please submit your issue directly in the Nextcloud Portal https://portal.nextcloud.com so it gets resolved more quickly by our dedicated engineers.
        Note that Nextcloud is an open source project backed by Nextcloud GmbH. Most of our volunteers are home users and thus primarily care about issues that affect home users. Our paid engineers prioritize issues of our customers. If you are neither a home user nor a customer, consider paying somebody to fix your issue, do it yourself or become a customer.
```

**Comment by tcitworld** [comment]
Testing current template at https://github.com/tcitworld/test/issues/new?assignees=&labels=bug%2C0.+to+triage&template=bug_report_form.yml

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    attributes:
+      label: Actual behaviour
+      description: |
+        Tell us what happens instead
+    validations:
+      required: true
+  - type: textarea
+    id: calendar-app-version
+    attributes:
+      label: calendar app version
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:36]:
```suggestion
      label: Calendar app version
```

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        Select Database engine serving Nextcloud Server.
+        _Describe in the "Additional info" section if you chose "Other"._
+      options:
+        - "MySQL"
+        - "MariaDB"
+        - "PostgreSQL"
+        - "SQlite"
+        - "Oracle"
+        - "Other"
+  - type: input
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:90]: This one should be a drop down too to pick between the two options

**Comment by GretaD** [comment]
thank you both for the review, please test the last changes here: https://github.com/GretaD/dummy-repository/blob/main/.github/ISSUE_TEMPLATE/bug_report_calendar.yml

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 6]
```diff
@@ -0,0 +1,167 @@
+name: "Calendar Bug"
+description: "Have you encountered a bug?"
+labels: ["bug", "0. to triage"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        ### 👍Thanks for reporting issues back to Nextcloud! 
+        This is the issue tracker of Nextcloud, if you have any support question please check out https://nextcloud.com/support
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:9]:
```suggestion
        This is the issue tracker of Nextcloud, if you have any support question please check out [the support page on our website](https://nextcloud.com/support). 
```

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        
+        This is the bug tracker for the web-calendar component. Find other components at https://github.com/nextcloud/
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:11]:
```suggestion
        This is the bug tracker for the web-calendar component. Find other components [in our Github Organisation](https://github.com/nextcloud/)
```

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        Everything that is related to the CalDAV server, meaning it does not only affect the web-calendar but also your mobile and desktop clients, belongs in the Server repository: https://github.com/nextcloud/server
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:12]:
```suggestion
        Everything that is related to the CalDAV server, meaning it does not only affect the web-calendar but also your mobile and desktop clients, belongs in [the Server repository](https://github.com/nextcloud/server).
```

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+        
+        For reporting potential security issues please see https://nextcloud.com/security/
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:14]:
```suggestion
        For reporting potential security issues please see [the security page on our website](https://nextcloud.com/security/).
```

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+        
+        To make it possible for us to help you please fill out below information carefully. 
+        You can also use the Issue Template application to prefill most of the required information: https://apps.nextcloud.com/apps/issuetemplate
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:17]:
```suggestion
        You can also use [the Issue Template application](https://apps.nextcloud.com/apps/issuetemplate) to prefill most of the required information.
```

Is this still recommended and up-to-date?

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+        
+        If you are a customer, please submit your issue directly in the Nextcloud Portal https://portal.nextcloud.com so it gets resolved more quickly by our dedicated engineers.
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:19]:
```suggestion
        If you are a customer, please submit your issue directly in [the Nextcloud Portal](https://portal.nextcloud.com) so it gets resolved more quickly by our dedicated engineers.
```

**Review by tcitworld — COMMENTED** [review]
Gave suggestions to improve links in markdown

File: .github/ISSUE_TEMPLATE/bug_report_form.yml (hunk 10 above)
Comment (tcitworld) [reply .github/ISSUE_TEMPLATE/bug_report_form.yml]: [Last supported version is NC 20](https://apps.nextcloud.com/apps/issuetemplate), so I'm guessing the sentence can be removed.

Comment (GretaD) [reply .github/ISSUE_TEMPLATE/bug_report_form.yml]: you said it :D

**Review by tcitworld — APPROVED** [review]

**Comment by GretaD** [comment]
hey @tcitworld i added also the feature form, sorry i forgot to add it before :)

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 12]
```diff
@@ -0,0 +1,37 @@
+name: "Feature request"
+description: "You have a neat idea that should be implemented?"
+labels: ["enhancement", "0. to triage"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        ### 👍Thanks for reporting issues back to Nextcloud!
```
Comment (tcitworld) [inline .github/ISSUE_TEMPLATE/feature_request.yml:8]: Add a space after `👍`.

Github doesn't seem to let me put more than one suggestion?

Comment (GretaD) [reply .github/ISSUE_TEMPLATE/feature_request.yml:8]: > Add a space after `👍`.
> 
> Github doesn't seem to let me put more than one suggestion?

write it as a comment. Just so you know, i only copy paste what we have on the old one, so give as many suggestions as you want :)

**Review by tcitworld — APPROVED** [review]

**Comment by miaulalala** [comment]
LGTML :shipit:

**Pull request merged**